openedx/core/djangoapps/user_api/management/commands/email_opt_in_list.py

```diff
@@ -111,22 +111,18 @@ def handle(self, *args, **options):
         if os.path.exists(file_path):
             raise CommandError("File already exists at '{path}'".format(path=file_path))
 
-        # Retrieve all the courses for the org.
-        # If we were given a specific list of courses to include,
-        # filter out anything not in that list.
-        courses = self._get_courses_for_org(org_list)
         only_courses = options.get("courses")
 
-        if only_courses is not None:
-            only_courses = [
-                CourseKey.from_string(course_key.strip())
-                for course_key in only_courses.split(",")
-            ]
-            courses = list(set(courses) & set(only_courses))
+        if only_courses is None:
+            # Retrieve all the courses for the org.
+            # If we were given a specific list of courses to include,
+            # filter out anything not in that list.
+            courses = self._get_courses_for_org(org_list)
 
-        # Add in organizations from the course keys, to ensure
-        # we're including orgs with different capitalizations
-        org_list = list(set(org_list) | set(course.org for course in courses))
+            # Add in organizations from the course keys, to ensure we're including orgs with different capitalizations
+            org_list = list(set(org_list) | set(course.org for course in courses))
+        else:
+            courses = list(set(only_courses.split(",")))
```

Comment:
Just so I understand the change here -- seems that only the "only_courses is not None" case has changed.  We have removed the "courses = list(set(courses) & set(only_courses))" filter as well as the CourseKey.from_string normalization and the augmentation of the org_list with any variants in the courses' org component.   If we never provide a list of courses, then this change would have no effect.   And if we do, we are basically trusting that list to be correct. Do we know which way we call this management command?

Reply:
That is correct.

We **always** supply a `--courses=<comma separate list>` argument, so for us this PR has a material impact.  I would think the set intersection filter is a good safety to have, but maybe we were concerned about a performance impact on the edxapp databases?  If that's the case, then this change is somewhat selfish, but I did also ask in #dev if there are any other users of this management command (I suspect it's only DE).

I'm also concerned about not calling CourseKey.from_string, but I guess whatever is the downstream consumer of that list can take either strings or CourseKeys?  It already is what we run in our weekly job so I assume it works---I plan on testing the analytics-email-optin-master job today anyway.

Reply:
Makes sense.   We presumably already get the course_ids from a call to the LMS, so I expect the change was instituted to provide speed and not run out of memory and that sort of thing.   I expect it's okay.